[SHT_LLVM_FUNC_ADDR_MAP] Introduce function address map section and emit dynamic instruction count

wlei-llvm opened the pull request.

wlei-llvm committed:
[SHT_LLVM_FUNC_ADDR_MAP] Introduce function address map section and emit dynamic instruction count feature

rlavaee commented:
Appreciate you doing this. I think we should now split this into at least three smaller PRs:

1. ObjectYaml changes which allow testing.
2. ELF support for encoding and decoding.
3. CodeGen changes for emitting the section.

wlei-llvm commented:
> Appreciate you doing this. I think we should now split this into at least three smaller PRs:
> 
> 1. ObjectYaml changes which allow testing.
> 2. ELF support for encoding and decoding.
> 3. CodeGen changes for emitting the section.

I split this into the following small PRs according to the suggestion! Thank you!

https://github.com/llvm/llvm-project/pull/124332 [SHT_LLVM_FUNC_MAP][ObjectYaml]Introduce function address map section and emit dynamic instruction count(ObjectYaml part)
https://github.com/llvm/llvm-project/pull/124333 [SHT_LLVM_FUNC_MAP][llvm-readobj]Introduce function address map section and emit dynamic instruction count(readobj part)
https://github.com/llvm/llvm-project/pull/124334 [SHT_LLVM_FUNC_MAP][CodeGen]Introduce function address map section and emit dynamic instruction count(CodeGen part)
